### PR TITLE
Add API docs link to each gallery demo

### LINKED ISSUE #21301: Let each Gallery demo link to their respective widget API doc

### PR DESCRIPTION
**Pull request by xster**
Fixes #21301

### CODE REVIEW

File: examples/flutter_gallery/lib/gallery/demos.dart [hunk 1]
```diff
@@ -510,3 +553,32 @@ final Map<GalleryDemoCategory, List<GalleryDemo>> kGalleryCategoryToDemos =
       return kAllGalleryDemos.where((GalleryDemo demo) => demo.category == category).toList();
     },
   );
+
+final Map<String, Widget> kDemoDocumentationMaterialButton =
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/gallery/demos.dart]: This seems like an odd way to provide the documentation button widgets. Why not:

```
StatelessWidget MaterialDemoDocumentationButton extends StatelessWidget {
  const MaterialDocumentationButton({
    @required String routeName,
  }) : assert(routeName != null), super(key: key);

  @override
  Widget build(BuildContext context) {
    GalleryDemo demo = kAllGalleryDemos.firstWhere((GalleryDemo demo) => demo.routeName == routeName);
    return IconButton(
      icon: const Icon(Icons.library_books),
      onPressed: () => launch(demo.documentationUrl, forceWebView: true)
    );
  }
}
```

Comment (xster) [reply examples/flutter_gallery/lib/gallery/demos.dart]: SG. Done.

File: examples/flutter_gallery/lib/demo/cupertino/cupertino_navigation_demo.dart [hunk 2]
```diff
@@ -125,6 +127,15 @@ class ExitButton extends StatelessWidget {
   }
 }
 
+Widget trailingButtons = Row(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/cupertino/cupertino_navigation_demo.dart]: final

File: examples/flutter_gallery/lib/gallery/demo.dart [hunk 3]
```diff
@@ -12,13 +15,15 @@ class ComponentDemoTabData {
     this.demoWidget,
     this.exampleCodeTag,
     this.description,
-    this.tabName
+    this.tabName,
+    this.documentationUrl,
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/gallery/demo.dart]: add this to operator ==, hashCode

File: examples/flutter_gallery/lib/gallery/demo.dart [hunk 4]
```diff
@@ -52,6 +57,13 @@ class TabbedComponentDemoScaffold extends StatelessWidget {
     }
   }
 
+  void _showApiDocumentation(BuildContext context) {
+    final String url = demos[DefaultTabController.of(context).index].documentationUrl;
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/gallery/demo.dart]: NICE

File: examples/flutter_gallery/lib/gallery/demo.dart [hunk 5]
```diff
@@ -170,3 +188,44 @@ class FullScreenCodeDialogState extends State<FullScreenCodeDialog> {
     );
   }
 }
+
+class MaterialDemoDocumentationButton extends StatelessWidget {
+  MaterialDemoDocumentationButton(String routeName, { Key key })
+    : documentationUrl = kDemoDocumentationUrl[routeName],
+      assert(
+        kDemoDocumentationUrl[routeName] != null,
+        'A documentation URL was not specified for this demo route in kAllGalleryDemos',
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/gallery/demo.dart]: for this demo route => for demo route "$routeName"

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**